### PR TITLE
[SPARK-22920][SPARKR] sql functions for current_date, current_timestamp, rtrim/ltrim/trim with trimString

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Add sql functions

## How was this patch tested?

manual, unit tests

### CODE REVIEW

File: R/pkg/R/functions.R [hunk 1]
```diff
@@ -2777,11 +2815,11 @@ setMethod("rpad", signature(x = "Column", len = "numeric", pad = "character"),
           })
 
 #' @details
-#' \code{substring_index}: Returns the substring from string str before count occurrences of
-#' the delimiter delim. If count is positive, everything the left of the final delimiter
-#' (counting from left) is returned. If count is negative, every to the right of the final
-#' delimiter (counting from the right) is returned. substring_index performs a case-sensitive
-#' match when searching for delim.
+#' \code{substring_index}: Returns the substring from string (\code{x}) before \code{count}
```
Comment (felixcheung) [inline R/pkg/R/functions.R]: this is mainly to fix the reference to `str` which doesn't exist here

File: R/pkg/DESCRIPTION [hunk 2]
```diff
@@ -59,3 +59,4 @@ Collate:
     'window.R'
 RoxygenNote: 5.0.1
 VignetteBuilder: knitr
+NeedsCompilation: no
```
Comment (felixcheung) [inline R/pkg/DESCRIPTION]: this gets flagged from CRAN when submitted

**Review by HyukjinKwon — APPROVED** [review]
LGTM

**Comment by felixcheung** [comment]
thanks!
merged to master

**Pull request closed without merging**